internal/npm_install/npm_install.bzl

```diff
@@ -375,24 +438,41 @@ def _add_scripts(repository_ctx):
         {},
     )
 
-def _workspace_root_path(repository_ctx, f):
-    segments = ["_"]
+# The directory in the external repository where we re-root the workspace by copying
+# package.json, lock file & data files to and running the package manager in the
+# folder of the package.json file.
+_WORKSPACE_REROOTED_PATH = "_"
+
+def _workspace_rerooted_path(repository_ctx, f):
+    segments = [_WORKSPACE_REROOTED_PATH]
     if f.package:
         segments.append(f.package)
     segments.append(f.name)
     return "/".join(segments)
 
-def _workspace_root_prefix(repository_ctx):
+def _workspace_root(repository_ctx):
```

Comment:
it's a bit confusing that this is a "workspace root" where it's actually a package.json dirname, right? is that an npm workspace, a bazel workspace, or likely neither?

Reply:
actually, this one happens to be the root of the user workspace which is where patches need to run from if symlink_node_modules is True. I'll ponder on better naming

Reply:
renamed the set of functions and added some comments to make clearer